js/components/todo-list.component.js

```diff
@@ -48,15 +56,28 @@ export class TodoListComponent extends AbstractComponent {
     window.addEventListener('add-task', this._dataChange.bind(this));
     window.addEventListener('update-tasks', this._dataChange.bind(this));
     window.addEventListener('delete-task', this._dataChange.bind(this));
+    window.addEventListener('edit-task', this._dataChange.bind(this));
+    window.addEventListener('show-active', this._dataChange.bind(this));
+    window.addEventListener('show-completed', this._dataChange.bind(this));
+    window.addEventListener('show-all', this._dataChange.bind(this));
+    window.addEventListener('clear-completed', this._dataChange.bind(this));
+    window.addEventListener('sort-by-text', this._dataChange.bind(this));
+    window.addEventListener('sort-by-date', this._dataChange.bind(this));
   }
 
   _afterCreate() {
     this.addEventListeners();
-    this._render();
+    this._render(this.tasks);
   }
 
-  _dataChange() {
-    this.tasks = taskData;
-    this._render();
+  _dataChange(event) {
+
+    if (event.detail) {
+      this.tasks = taskData;
+      this._render(event.detail);
+    }else {
+      this.tasks = taskData;
+      this._render(this.tasks);
```

Comment:
const itemToRender = event.detail ? event.detail : this.tasks

this.tasks = taskData
this._render(itemToRender)